### PR TITLE
Pract.05: Add 07-09 | Miroslav Stoyanovski

### PR DESCRIPTION
**Pull request by mir4oni3**
Miroslav Stoyanovski
6 5MI0600195

### CODE REVIEW

File: SI/Sem.05/Pract.05/Task09.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		if (turn) {
+			std::cout << "X turn" << std::endl;
+		}
+		else {
+			std::cout << "O turn" << std::endl;
+		}
+		int row, col;
+		std::cin >> row >> col;
+		if (row > 3 || col > 3) {
+			continue;
```
Comment (GeorgiTerziev02) [inline SI/Sem.05/Pract.05/Task09.cpp:31]: Check row or col < 0

File: SI/Sem.05/Pract.05/Task09.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		//check for draw
+		bool draw = true;
+		for (int i = 0; i < 3; i++) {
+			for (int j = 0; j < 3; j++) {
+				if (values[i][j] == ' ') {
+					draw = false;
+					break;
+				}
+			}
+		}
```
Comment (GeorgiTerziev02) [inline SI/Sem.05/Pract.05/Task09.cpp:48]: create function

File: SI/Sem.05/Pract.05/Task09.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+				won = true;
+				break;
+			}
+			if (values[1][1] != ' ' && ((values[0][0] == values[1][1] && values[1][1] == values[2][2]) ||
+				(values[0][2] == values[1][1] && values[1][1] == values[2][0]))) {
+				print(values);
+				std::cout << values[1][1] << " won!" << std::endl;
+				won = true;
+				break;
+			}
```
Comment (GeorgiTerziev02) [inline SI/Sem.05/Pract.05/Task09.cpp:74]: Duplicate code

File: SI/Sem.05/Pract.05/Task09.cpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
+#include <iostream>
+
+void print(char values[3][3]) {
```
Comment (GeorgiTerziev02) [inline SI/Sem.05/Pract.05/Task09.cpp:3]: Use these two cool lines to clear your console before every print
    std::cout << "\033[;H"; // Moves cursor to the top left
    std::cout << "\033[J"; // Clears the console

File: SI/Sem.05/Pract.05/Task09.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		}
+		int row, col;
+		std::cin >> row >> col;
+		if (row > 3 || col > 3) {
+			continue;
+		}
+		row--;
+		col--;
+		if (values[row][col] == ' ') {
+			values[row][col] = turn ? values[row][col] = 'X' : values[row][col] = 'O';
```
Comment (GeorgiTerziev02) [inline SI/Sem.05/Pract.05/Task09.cpp:36]: You are assigning the value two times here!

**Review by GeorgiTerziev02 — CHANGES_REQUESTED** [review]
Looks good, Good job! It would be great if you fix my comments. 👏 👏 👏

**Comment by Justsvetoslavov** [comment]
Refactor the code and you can put a PR about tic-tac-toe as separate.

**Pull request closed without merging**